components/ActivityForm/index.js

```diff
@@ -2,6 +2,123 @@ import styled from "styled-components";
 import { useRouter } from "next/router";
 import categories from "@/assets/categories";
 import { useState } from "react";
+import Link from "next/link";
+import { StyledButton } from "../ActivityDetails";
+
+export default function ActivityForm({ activity, onSubmit, isUpdateMode }) {
+  const [selectedCategoryIds, setSelectedCategoryIds] = useState(
+    activity?.categoryIds || []
+  );
+
+  const router = useRouter();
+
+  function handleChange(event) {
+    const categoryId = event.target.value;
+    setSelectedCategoryIds(
+      selectedCategoryIds.includes(categoryId)
+        ? selectedCategoryIds.filter((id) => id !== categoryId)
+        : [...selectedCategoryIds, categoryId]
+    );
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formElements = new FormData(event.target);
+    const newActivity = Object.fromEntries(formElements);
+    newActivity.categoryIds = selectedCategoryIds;
```

Comment:
```suggestion
    const formData = new FormData(event.target);
    const activityData = Object.fromEntries(formData);
    
    const newActivity = { ...activityData, categoryIds: selectedCategoryIds };
```

Well done using the FormData object! 💪
You could even write this in a more strict functional manner by not mutating _newActivity_